includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2220,6 +2220,14 @@ public function get_level3_data_from_order( WC_Order $order ): array {
 			$tax_amount      = WC_Payments_Utils::prepare_amount( $item->get_total_tax(), $currency );
 			$discount_amount = WC_Payments_Utils::prepare_amount( $subtotal - $item->get_total(), $currency );
 
+			// This is a special case, to prevent Stripe error when $unit_cost is negative. The unit cost can be
+			// negative if the item has a negative price.
+			if ( 0 > $subtotal ) {
```

Comment:
In this case, we should rather refactor the logic than try to patch it.
The used approach will work, but pretty quick context is lost and it'll be no clue if this block can be relocated.

To make it maintainable, the code structure shall be changed in the way to say "this is part of discount calculation, not a randomly placed condition."

In the practical field:

```php
			$description     = substr( $item->get_name(), 0, 26 );
			$quantity          = ceil( $item->get_quantity() );
			$tax_amount    = WC_Payments_Utils::prepare_amount( $item->get_total_tax(), $currency );
			if ( $subtotal >= 0 ) {
				$unit_cost             = WC_Payments_Utils::prepare_amount( $subtotal / $quantity, $currency );
				$discount_amount = WC_Payments_Utils::prepare_amount( $subtotal - $item->get_total(), $currency );
			} else {
				// It's possible to create products with negative price - represent it as free one with discount.
				$discount_amount = abs( WC_Payments_Utils::prepare_amount( $subtotal / $quantity, $currency ) );
				$unit_cost              = 0;
			}
```

Reply:
Nice improvement thanks for suggesting!

Reply:
Addressed in [d72732eab99cb897cf6bb4a1ffebee16c4e6b257](https://github.com/Automattic/woocommerce-payments/pull/4244/commits/d72732eab99cb897cf6bb4a1ffebee16c4e6b257)